cdk-infra/utils/buckets.ts

```diff
@@ -10,3 +16,149 @@ export const docsBucketNames = [
   'docs-java',
   'docs-node',
 ] as const;
+
+interface CustomBucketProps {
+  scope: Construct;
+  featureName: string;
+  env: AutoBuilderEnv;
+  bucketName: string;
+}
+export function createCustomBucket({ scope, featureName, env, bucketName }: CustomBucketProps): Bucket {
+  const stackBucketName = `${featureName}-${bucketName}-${env}`.toLowerCase();
+  const websiteRoutingRules = bucketName === 'docs-mongodb-org' ? getMongoDocsBucketRoutingRules(env) : undefined;
```

Comment:
It is very possible we'll have custom routing rules (e.g. bucket-level redirects) for other buckets at some point -- will we need to change this when that happens?

Reply:
Yep! I think we can refactor how we get the redirect rules in the future to make them specific to a bucket, but I don't believe that should be too complicated. I can brainstorm ways in which we can handle different websiteRoutingRules for a given bucket. Maybe I could use a map of sorts.

Reply:
Sweet, this doesn't need to be blocking, I just wanted to double check feasibility.